### ISSUE TITLE
QueryArgument Bug:  Can't instantiate with ResolvedType, only System.Type

### ISSUE THREAD

**Issue by isaksky** [graphql-dotnet/graphql-dotnet#387]
The source for QueryArgument reads:

```csharp
public class QueryArgument : IHaveDefaultValue {
        public QueryArgument(Type type) {
            if (type == null || !typeof(GraphType).IsAssignableFrom(type)) {
                throw new ArgumentOutOfRangeException(nameof(type), "QueryArgument type is required and must derive from GraphType.");
            }
            Type = type;
        }
        public string Name { get; set; }
        public string Description { get; set; }
        public object DefaultValue { get; set; }
        public IGraphType ResolvedType { get; set; }
        public Type Type { get; private set; }
    }
```

The problem here is you get an exception when building a schema dynamically:

```csharp
var qa = new QueryArgument(null); // explodes
qa.ResolvedType = myResolvedType; // otherwise, would have done this after.
```

If I create it with a false type (e.g., typeof(IntGraphType)), then set ResolvedType to what I want, then nullify `.Type` with reflection, I get an invalid schema (GraphiQL complains **), so the error seems to not be as shallow as only being in the `QueryArgument` class.

Related (closed) issue: https://github.com/graphql-dotnet/graphql-dotnet/issues/206 .

** GraphiQL JS Error with hack, in case it helps:

```
Error: Introspection must provide input type for arguments.
    at exports.default (https://cdnjs.cloudflare.com/ajax/libs/graphiql/0.11.2/graphiql.min.js:1:555526)
    at getInputType (https://cdnjs.cloudflare.com/ajax/libs/graphiql/0.11.2/graphiql.min.js:1:673828)
    at buildInputValue (https://cdnjs.cloudflare.com/ajax/libs/graphiql/0.11.2/graphiql.min.js:1:677522)
    at https://cdnjs.cloudflare.com/ajax/libs/graphiql/0.11.2/graphiql.min.js:1:556376
    at Array.reduce (<anonymous>)
    at exports.default (https://cdnjs.cloudflare.com/ajax/libs/graphiql/0.11.2/graphiql.min.js:1:556326)
    at buildInputValueDefMap (https://cdnjs.cloudflare.com/ajax/libs/graphiql/0.11.2/graphiql.min.js:1:677375)
    at https://cdnjs.cloudflare.com/ajax/libs/graphiql/0.11.2/graphiql.min.js:1:677239
    at https://cdnjs.cloudflare.com/ajax/libs/graphiql/0.11.2/graphiql.min.js:1:556376
    at Array.reduce (<anonymous>)
```

**Comment by joemcbride** [comment]
Thanks for the report, I ran into this same issue.  I have this fixed in this PR: https://github.com/graphql-dotnet/graphql-dotnet/pull/362

**Comment by joemcbride** [comment]
This fix is available in the 2.0 alpha release.
